plugins/osdn/registry.go

```diff
@@ -666,3 +671,39 @@ EndpointLoop:
 
 	registry.baseEndpointsHandler.OnEndpointsUpdate(filteredEndpoints)
 }
+
+func (registry *Registry) trackPod(pod *kapi.Pod) {
+	if pod.Status.PodIP != "" {
+		info, ok := registry.namespaceOfPodIP[pod.Status.PodIP]
+
+		if pod.Status.Phase == kapi.PodPending || pod.Status.Phase == kapi.PodRunning {
+			// When a pod hits one of the states where the IP is in use then
+			// we need to add it to our IP -> namespace tracker.  There _should_ be no
+			// other entries for the IP if we caught all of the right messages, so warn
+			// if we see one, but clobber it anyway since the IPAM
+			// should ensure that each IP is uniquely assigned to a pod (when running)
+			if ok && info.uid != pod.UID {
+				log.Warningf("IP '%s' was marked as used by namespace '%s' (pod '%s')... updating to namespace '%s' (pod '%s')",
+					pod.Status.PodIP, info.namespace, info.uid, pod.ObjectMeta.Namespace, pod.UID)
+			}
+
+			registry.namespaceOfPodIP[pod.Status.PodIP] = podInfo{namespace: pod.ObjectMeta.Namespace, uid: pod.UID}
```

Comment:
maybe just store the *kapi.Pod itself for simplicity? Also, "namespaceOfPodIP" is no longer an accurate description of the data...